### ISSUE TITLE
Improve `doc/internals/contributing.rst`

### ISSUE THREAD

**Issue by ezio-melotti** [sphinx-doc/sphinx#10437]
**Is your feature request related to a problem? Please describe.**
I followed the [Contributing to Sphinx](https://www.sphinx-doc.org/en/master/internals/contributing.html) docs and some parts were not clear enough and could be improved.

**Describe the solution you'd like**
The following parts of the [Getting started section](https://www.sphinx-doc.org/en/master/internals/contributing.html#getting-started) should be improved/updated.

* "4.Checkout the appropriate branch." is not too clear.  It says to use `A.x` (what's `A`?) for backward-compatible changes, and `master` for incompatible/substantial changes.  I want to submit a bug fix that, I think, should go on all active branches.  Should I make the PR against `5.x` (i.e. the highest "`A`")?  Will it be back/forward-ported to `4.x`/`master`?  Is `4.x` still receiving bug fixes?  There is also a link to the [Sphinx’s release process](https://www.sphinx-doc.org/en/master/internals/release-process.html), but it's in a sentence about "urgent release", so it doesn't seem related.
* "5. Setup a virtual environment." recommends to use `virtualenv ~/.venv` -- wouldn't `python3 -m venv .venv` be better?  (`.venv` is also already included in the `.gitignore` file).  The description is also not particularly clear: should I created it if I want to be able to run tests without making global changes to my environment (including installing dependencies)?
* "6. Create a new working branch." could recommend the more modern alternative `git switch -c feature-xyz` instead (or in addition, for backward compatibility) to the old one.
* "7. Hack, hack, hack." recommends to start hacking, but I think it would make sense to run the tests first to see if they pass before breaking them.
* "8. Add a bullet point to CHANGES ..." should link to the file (and/or say "to the CHANGES file").  This section also suggests to add an issue number to the commit message.  Should an issue always be created before a PR?  Should I mention the issue number in all commits I do in the branch?

The [Unit tests](https://www.sphinx-doc.org/en/master/internals/contributing.html#unit-tests) section seems to assume that I have `tox` installed and lists several commands to test in specific scenarios.  What's the simplest set of commands needed to install the necessary dependencies in a venv and run all tests?  Do I need `tox` or is `pytest` enough?  How do I install/run it?

Eventually I was able to figure out most of the answers, but it would be better if they were documented clearly.

**Describe alternatives you've considered**
Giving up contributing to Sphinx or submitting an incomplete PR without tests :upside_down_face: 

**Additional context**
I wanted to fix an issue I found, and I was following the docs in order to create a PR.

**Comment by AA-Turner** [comment]
I agree, I'll have a go at writing up a new wording & will ping you for a review.

A

**Comment by tk0miya** [comment]
+1. I'm not familiar with English. So it's very difficult task to update the document to me. Your comment is very welcome :-)

**Comment by jfbu** [comment]
> 
> 
> The [Unit tests](https://www.sphinx-doc.org/en/master/internals/contributing.html#unit-tests) section seems to assume that I have `tox` installed and lists several commands to test in specific scenarios. What's the simplest set of commands needed to install the necessary dependencies in a venv and run all tests? Do I need `tox` or is `pytest` enough? How do I install/run it?
> 
> Eventually I was able to figure out most of the answers, but it would be better if they were documented clearly.
> 

Perhaps this section could as alternative to `tox` mention 
- doing `pip install -e .[test,docs]` and advise also building Sphinx own docs as extra test 
- have a look at `setup.py` to learn about dependencies installed by the above or other options,
- mention that on Unixen `make test` works to launch the test suite with `make test TEST=tests/test_build_latex.py`  and `make test TEST=tests/test_ext_autodoc.py::test_parse_name` as possible syntax examples to run only selected ones.

**Comment by ezio-melotti** [comment]
> * doing `pip install -e .[test,docs]` and advise also building Sphinx own docs as extra test

After running this, `make test` worked.  This is also mentioned after the `tox` section, but maybe it should be mentioned earlier.  In the "Getting started" section, at step 5, it suggest to run `pip install -e .`, but this doesn't include the tests and their dependencies.  Perhaps an option to install everything could be mentioned there.

> * mention that on Unixen `make test` works to launch the test suite

I would also mention `pytest -v`, which should work on all platforms and it might be simpler/more familiar to users (especially compared to the `make test TEST=...` syntax).

IOW, from this section I would expect, in order:
1. whether I need to install any additional dependencies, and the command to install them
2. a simple command to run all the tests (such as `pytest -v` or `make test`)
3. more advanced commands to run specific or additional tests
4. alternative options such as `tox` (note that even after `pip install -e .[test,docs]`, `tox` is still not installed)

**Comment by AA-Turner** [comment]
`make test TEST=...` also doesn't work on Windows.

A
